ocaml/xapi/import_raw_vdi.ml

```diff
@@ -114,13 +114,24 @@ let import vdi (req: Request.t) (s: Unix.file_descr) _ =
               then localhost_handler rpc session_id vdi req s
               else
                 let host = Importexport.find_host_for_sr ~__context sr in
-                let address = Db.Host.get_address ~__context ~self:host in
-                return_302_redirect req s address;
-                None
+                if host <> Ref.null then
+                  let address = Db.Host.get_address ~__context ~self:host in
+                  return_302_redirect req s address;
+                else begin
+                  Xapi_http.with_context "VDI.import" req s
+                    (fun __context ->
+                      TaskHelper.failed ~__context Api_errors.(Server_error(no_hosts_available,[]));
+                      Http_svr.headers s (Http.http_400_badrequest ());
+                    )
+                end;
+              None
             | None ->
               error "Require an SR or VDI to import";
-              TaskHelper.failed ~__context Api_errors.(Server_error(vdi_missing,[]));
-              Http_svr.headers s (Http.http_400_badrequest ());
+              Xapi_http.with_context "VDI.import" req s
+                (fun __context ->
+                  TaskHelper.failed ~__context Api_errors.(Server_error(vdi_missing,[]));
+                  Http_svr.headers s (Http.http_400_badrequest ())
+                );
```

Comment:
Yes, will do it. My initial thought of this change is to get the task_id from the request.
Thanks